src/main/java/com/auth0/json/auth/TokenHolder.java

```diff
@@ -1,36 +1,41 @@
 package com.auth0.json.auth;
 
-import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
-import com.fasterxml.jackson.annotation.JsonInclude;
-import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import java.util.Date;
 
 /**
  * Class that contains the Tokens obtained after a call to the {@link com.auth0.client.auth.AuthAPI} methods.
  */
 @SuppressWarnings("unused")
-@JsonIgnoreProperties(ignoreUnknown = true)
-@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonDeserialize(using = TokenHolderDeserializer.class)
 public class TokenHolder {
 
-    @JsonProperty("access_token")
     private String accessToken;
-    @JsonProperty("id_token")
     private String idToken;
-    @JsonProperty("refresh_token")
     private String refreshToken;
-    @JsonProperty("token_type")
     private String tokenType;
-    @JsonProperty("expires_in")
     private long expiresIn;
-    @JsonProperty("scope")
     private String scope;
+    private Date expiresAt;
+
+    public TokenHolder() {}
```

Comment:
As the current implementation does not provide a constructor, adding a new constructor without a default constructor could be considered a breaking change (not likely encountered as the `TokenHolder` is specific to a response from the Auth0 server, but could be used in tests, better safe than sorry)

Reply:
Should we annotate this with suppress unused, with a comment about this that you are saying above? for future devs

Reply:
Updated per recommendation.